instrumentation/messaging/src/main/java/brave/messaging/MessagingParser.java

```diff
@@ -0,0 +1,43 @@
+package brave.messaging;
+
+import brave.SpanCustomizer;
+import brave.propagation.TraceContext;
+import brave.propagation.TraceContextOrSamplingFlags;
+
+public class MessagingParser {
+
+  public <Chan, Msg> void message(ChannelAdapter<Chan> channelAdapter,
+      MessageAdapter<Msg> messageAdapter,
+      Chan channel, Msg message, SpanCustomizer customizer) {
+    customizer.name(messageAdapter.operation(message));
+    channel(channelAdapter, channel, customizer);
+    identifier(messageAdapter, message, customizer);
+  }
+
+  public <Chan> void channel(ChannelAdapter<Chan> adapter, Chan chan,
+      SpanCustomizer customizer) {
+    String channel = adapter.channel(chan);
+    if (chan != null) customizer.tag(adapter.channelTagKey(chan), channel);
+  }
+
+  public <Msg> void identifier(MessageAdapter<Msg> adapter, Msg message,
+      SpanCustomizer customizer) {
+    String identifier = adapter.identifier(message);
+    if (identifier != null) {
```

Comment:
my 2p is that the type of identifier should be its own value. for example, if it is important that the type is correlation id. "message.idenifier_type=correlation" In this way, FinishedSpanHandler can choose to take whatever the identifier is without having to first see what the key could be.. for example, it might never know the name of the key returned here. Stable keys can help with portable data policy iotw